classes/Dispatcher.php

```diff
@@ -373,20 +373,19 @@ public function dispatch()
 
             // Dispatch module controller for front office
             case self::FC_MODULE:
-                $module_name = Validate::isModuleName(Tools::getValue('module')) ? Tools::getValue('module') : '';
-                $module = Module::getInstanceByName($module_name);
+                $module = Validate::isModuleName(Tools::getValue('module')) ? Module::getInstanceByName(Tools::getValue('module')) : false;
                 $controller_class = 'PageNotFoundController';
                 if (Validate::isLoadedObject($module) && $module->active) {
```

Comment:
As `$module` can now be false I think we can edit the following condition for easier understanding

```suggestion
                if ($module !== false && Validate::isLoadedObject($module) && $module->active) {
```